bluecellulab/analysis/analysis.py

```diff
@@ -57,47 +92,88 @@ def compute_plot_iv_curve(cell, stim_start=100.0, duration=500.0, post_delay=100
+                          stim_start=100.0,
+                          duration=500.0,
+                          post_delay=100.0,
+                          max_current=0.8,
+                          nb_bins=11):
+    """Compute and plot the Frequency-Current (F-I) curve for a given cell by
+    injecting a range of currents.
+
+    This function evaluates the relationship between injected current amplitude and the firing rate
+    of a neuronal cell model. Currents are injected at a specified section and segment, and the number
+    of spikes recorded in the specified recording location is used to construct the F-I curve.
 
     Args:
-        cell (bluecellulab.cell.Cell): The initialized cell model.
-        stim_start (float): Start time for current injection (in ms). Default is 100.0 ms.
-        duration (float): Duration of current injection (in ms). Default is 500.0 ms.
-        post_delay (float): Delay after the stimulation ends (in ms). Default is 100.0 ms.
-        max_current (float): Maximum current to inject (in nA). Default is 0.8 nA.
-        nb_bins (int): Number of current injection levels. Default is 11.
+        cell (bluecellulab.cell.Cell): The initialized BlueCelluLab cell model.
+        injecting_section (str, optional): The name of the section where the stimulus is injected.
+            Default is "soma[0]".
+        injecting_segment (float, optional): The position along the injecting section (0.0 to 1.0)
+            where the stimulus is applied. Default is 0.5.
+        recording_section (str, optional): The name of the section where spikes are recorded.
+            Default is "soma[0]".
+        recording_segment (float, optional): The position along the recording section (0.0 to 1.0)
+            where spikes are recorded. Default is 0.5.
+        stim_start (float, optional): The start time of the current injection (in ms). Default is 100.0 ms.
+        duration (float, optional): The duration of the current injection (in ms). Default is 500.0 ms.
+        post_delay (float, optional): The delay after the stimulation ends before the simulation stops
+            (in ms). Default is 100.0 ms.
+        max_current (float, optional): The maximum amplitude of the injected current (in nA).
+            Default is 0.8 nA.
+        nb_bins (int, optional): The number of discrete current levels between 0 and `max_current`.
+            Default is 11.
 
     Returns:
         tuple: A tuple containing:
-            - list_amp (np.ndarray): The predefined injected step current levels (nA).
-            - spike_count (np.ndarray): The corresponding spike counts.
+            - list_amp (np.ndarray): The injected current amplitudes (nA).
+            - spike_count (np.ndarray): The corresponding spike counts for each current amplitude.
+
+    Raises:
+        ValueError: If the cell object is invalid or the specified sections/segments are not found.
     """
 
     rheobase = calculate_rheobase(cell)
```

Comment:
If the injecting_section != soma, then the rheobase value will still be that of soma here. Same here iv curve function.